tracing-subscriber/src/subscribe/mod.rs

```diff
@@ -686,7 +686,31 @@ pub(crate) mod tests;
 /// be composed together with other subscribers to build a [collector]. See the
 /// [module-level documentation](crate::subscribe) for details.
 ///
+/// # Enabling interest
+///
+/// Whenever an tracing event (or span) is emitted, it goes through a number of
+/// steps to determine how and how much it should be processed. The earlier an
+/// event is disabled, the less work has to be done to process the event, so
+/// subscribers should attempt to provide accurate information as early as
+/// possible. Note that this determines **global** interest for the entire stack
+/// of subscribers; a subscriber that wants to ignore certain events/spans
+/// should just ignore them in the notifications. In order, each event checks:
+///
+/// - [`register_callsite`], once per callsite (roughly: once per time that
+///   `event!` or `span!` is written in the source code; this is cached at the
+///   callsite). See [`Collect::register_callsite`] for how this behaves.
+/// - [`enabled`], once per emitted event (or span). This is the main point to
```

Comment:
i think the language "once per emitted event" could maybe be clarified a little bit --- I understand what you mean here, but it could be good to state as clearly as possible what the distinction between `enabled` and `register_callsite` is...

Reply:
also, we should note that `enabled` is _only_ called if `register_callsite` returns `Interest::sometimes`; if the interest is `always`, `enabled` is skipped and the callsite is always recorded; and if it's `never`, `enabled` is not called as the callsite is skipped.

Reply:
👍